### ISSUE TITLE
Role fails if DNS domain is not set on the managed node

### ISSUE THREAD

**Issue by alexxa** [linux-system-roles/sap-preconfigure#98]
This is a ToDo note to check on workaround for the issue when the roles fails if DNS domain is not set on the managed node, and a user needs to specify it and rerun the play.

Intentionally, Ansible doesn't allow to prompt users on tasks level, but it seems it's possible to do in a pre-task using pause + prompt + set_facts + when (dns is missing). Another solution may be an action plugin to ask the user to input a variable (or interrupt the play).

**Comment by berndfinger** [comment]
The RHEL System Roles for SAP document for version 2 will contain the following text:
In case there is no DNS domain set on the managed node, which is typically the case on cloud systems, the role sap-preconfigure will fail in task "Verify that the DNS domain is set". To avoid this, set variable sap_domain in file /usr/share/ansible/roles/sap-preconfigure/defaults/main.yml or run the ansible-playbook command with line parameter
-e "sap_domain=example.com" (with the domain name being example.com in this case - please replace it by your domain name).
Is that OK for you?

**Comment by alexxa** [comment]
Yes, sure. It's good to publish it as a workaround. However, I would recommend a fix for the next version as per initial comment.

**Comment by berndfinger** [comment]
Closing after copying over to https://github.com/sap-linuxlab/community.sap_install/issues/3.
